### PR TITLE
Adding Identity Extensions Implementation

### PR DESCRIPTION
**Pull request by Ashi1993**
# Adding Identity Extensions Implementation

> This PR adds the Identity Extensions Implementation.

**Issue link:** *https://github.com/wso2-enterprise/solutions-bfsi/issues/20*

**Doc Issue:** *Optional, link issue from [documentation repository](https://github.com/wso2/docs-ob/issues)*

**Applicable Labels:** *Spec, product, version, type (specify requested labels)*

------

### Development Checklist

1. [x] Build complete solution with pull request in place.
2. [x] Ran checkstyle plugin with pull request in place.
3. [x] Ran Findbugs plugin with pull request in place.
4. [x] Ran FindSecurityBugs plugin and verified report.
5. [x] Formatted code according to WSO2 code style.
6. [x] Have you verified the PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?
7. [ ] Migration scripts written (if applicable).
8. [x] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [ ] Written unit tests.
2. [ ] Verified tests in multiple database environments (if applicable).
3. [ ] Tested with BI enabled  (if applicable).

### CODE REVIEW

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ */
+public class BFSIDefaultAuthServletImpl implements BFSIAuthServletInterface {
+
+    private String jspPath;
+    @Override
+    public Map<String, Object> updateRequestAttribute(HttpServletRequest request, JSONObject dataSet,
+                                                      ResourceBundle resourceBundle) {
+
+        String consentType = dataSet.getString("type");
+        switch (consentType) {
```
Comment (VimukthiRajapaksha) [inline financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java:45]:
```suggestion
        switch (dataSet.getString(constants.TYPE)) {
```

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+            case ConsentExtensionConstants.ACCOUNTS:
+                return Utils.populateAccountsData(request, dataSet);
+            case ConsentExtensionConstants.PAYMENTS:
+                return Utils.populatePaymentsData(request, dataSet);
+            case ConsentExtensionConstants.FUNDS_CONFIRMATIONS:
+                return Utils.populateCoFData(request, dataSet);
+            default:
+                return new HashMap<>();
```
Comment (VimukthiRajapaksha) [inline financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java:54]:
```suggestion
                return CollectionUtils.emptyMap();
```

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        }
+    }
+
+    @Override
+    public Map<String, Object> updateSessionAttribute(HttpServletRequest request, JSONObject dataSet,
+                                                      ResourceBundle resourceBundle) {
+
+        return new HashMap<>();
```
Comment (VimukthiRajapaksha) [inline financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java:62]: Applicable for other places as well. 

```suggestion
        return CollectionUtils.emptyMap();
```

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    // Suppressed warning count - 1
+    public Map<String, Object> updateConsentData(HttpServletRequest request) {
+
+        Map<String, Object> returnMaps = new HashMap<>();
+
+        String[] accounts = request.getParameter("accounts[]").replaceAll("[\r\n]", "").split(":");
+        returnMaps.put("accountIds", List.of(accounts));
+
+        returnMaps.put(ConsentExtensionConstants.PAYMENT_ACCOUNT,
+                request.getParameter(ConsentExtensionConstants.PAYMENT_ACCOUNT).replaceAll("[\r\n]", ""));
```
Comment (VimukthiRajapaksha) [inline financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java:79]: Shall we move `"[\r\n]"` to a constant? and consider using StringUtils.empty() where applicable.

Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java]: Fixed in https://github.com/wso2/financial-services-accelerator/pull/109/commits/c8dc488ccfe9eb51e8e5ed3fc3b63013f10843d0

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java (hunk 3 above)
Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java]: Fixed in https://github.com/wso2/financial-services-accelerator/pull/109/commits/c8dc488ccfe9eb51e8e5ed3fc3b63013f10843d0

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java (hunk 2 above)
Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java]: Fixed in https://github.com/wso2/financial-services-accelerator/pull/109/commits/c8dc488ccfe9eb51e8e5ed3fc3b63013f10843d0

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java (hunk 1 above)
Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java]: Fixed in https://github.com/wso2/financial-services-accelerator/pull/109/commits/c8dc488ccfe9eb51e8e5ed3fc3b63013f10843d0

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.ResourceBundle;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * The default implementation for BFSI flow.
+ */
+public class BFSIDefaultAuthServletImpl implements BFSIAuthServletInterface {
+
+    private String jspPath;
```
[the remark this answers is not in the document]
Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authservlet/impl/BFSIDefaultAuthServletImpl.java]: Fixed in https://github.com/wso2/financial-services-accelerator/pull/109/commits/c8dc488ccfe9eb51e8e5ed3fc3b63013f10843d0

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/util/FinancialServicesUtils.java [hunk 6]
```diff
@@ -119,4 +122,44 @@ public static String reduceStringLength(String input, int maxLength) {
...
+     * @param jwtToken jwt sent by the tpp
+     * @param jwtPart  expected jwt part (header, body)
+     * @return json object containing requested jwt part
+     * @throws java.text.ParseException if an error occurs while parsing the jwt
+     */
+    public static JSONObject decodeRequestJWT(String jwtToken, String jwtPart) throws java.text.ParseException {
+
+        JWSObject plainObject = JWSObject.parse(jwtToken);
+
+        if ("head".equals(jwtPart)) {
```
[the remark this answers is not in the document]
Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/util/FinancialServicesUtils.java]: Fixed in https://github.com/wso2/financial-services-accelerator/pull/109/commits/c8dc488ccfe9eb51e8e5ed3fc3b63013f10843d0

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/internal/ConsentExtensionsServiceComponent.java [hunk 7]
```diff
@@ -74,7 +75,8 @@ protected void deactivate(ComponentContext context) {
     }
 
     @Reference(
-            service = org.wso2.financial.services.accelerator.consent.mgt.service.ConsentCoreService.class,
+            name = "ConsentCoreService",
```
Comment (VimukthiRajapaksha) [inline financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/internal/ConsentExtensionsServiceComponent.java:78]:
```suggestion
            name = ConsentCoreService.class.getName(),
```

**Review by VimukthiRajapaksha — APPROVED** [review]

**Review by RivinduM — APPROVED** [review]

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/internal/ConsentExtensionsServiceComponent.java (hunk 7 above)
Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/internal/ConsentExtensionsServiceComponent.java]: Cannot use this as the name hence reverting back

**Pull request merged**